include/picongpu/fields/incidentField/profiles/FromOpenPMDPulse.hpp

```diff
@@ -594,52 +611,76 @@ namespace picongpu
                          */
                         HDINLINE float_X linInterpol(
                             typename pmacc::Buffer<float_X, 3u>::DataBoxType const& dataBox,
-                            float3_X const& pos) const
+                            floatD_X const& pos) const
                         {
                             // find the index in the data box which is nearest to pos
-                            DataSpace<3u> const idxClosest
-                                = static_cast<pmacc::math::Vector<int, 3u>>(pos + float3_X::create(0.5_X));
+                            DataSpace<3u> idxClosest;
+                            for(uint32_t d = 0u; d < 3u; d++)
+                            {
+                                if(d == 2u and simDim == DIM2)
+                                    /* for 2D simulations, we only take the central slice of the field data parallel to
+                                     * propagation and polarisation direction. Hence, the index accessing the second
+                                     * transverse direction (d = 2) remains constant.
+                                     */
+                                    idxClosest[d] = static_cast<uint>(extentOpenPMDdataBox(d) / 2._X);
+                                else
+                                    // propagation and polarisation direction are treated normally
+                                    idxClosest[d] = static_cast<uint>(pos[d] + 0.5_X);
+                            }
 
-                            // the other 7 nearest neighbour indices still have to be found
-                            DataSpace<3u> idxShift; // shift to the other nearest neighbour indices
+                            // the other 7 (in 3D) / 3 (in 2D) nearest neighbour indices still have to be found
+                            DataSpace<3u> idxShift; // shift from idxClosest to idxNext
                             DataSpace<3u> idxNext; // nearest neighbour index
                             float3_X weight;
-                            for(uint32_t d = 0u; d < 3u; d++)
+                            for(uint32_t d = 0u; d < simDim; d++)
                             {
-                                if(idxClosest[d] == 0) // to avoid border problems
-                                    idxShift[d] = 1;
+                                if(idxClosest[d] == 0u) // to avoid border problems
```

Comment:
In order to avoid that this comment ever shifts to the next line and becomes detached from what it points to:
```suggestion
                                // to avoid border problems
                                if(idxClosest[d] == 0u)
```

Also, just for my information: What is the edge case you're guarding against here? It looks like the `else if` branch could never be taken (assuming `pos` to be positive), so this explicit branch doesn't seem to be necessary. And if there is a problem: Why don't we have to guard the "upper" edge case as well?

Reply:
The upper edge case is included in the `else if` branch. And both the `else if` and the `else` brach are the standard scenario used in 99.9% of the cases. `pos` is $\geqslant 0$ as you said, but it is `float_X` and `pos[d] - static_cast<float_X>(idxClosest[d])` can be both greater or smaller than 0, depending on the position of `pos` with respect to the grid points of `dataBox`. I hope I expressed myself somehow understandable. If not, please ask again!